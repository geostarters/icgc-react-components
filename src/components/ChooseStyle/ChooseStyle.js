// @flow

import React from "react";
import PropTypes from "prop-types";

import { Image } from "semantic-ui-react";

import styles from "./ChooseStyle.module.css";

const ChooseStyle = ({currentStyleId, stylesList, handleClickStyle, selectedColor, labelColor}) => (<div className={styles.styleContainer}>
	<ul>
		{stylesList.map((bg, index) => <li key={index} onClick={e => (currentStyleId === bg.styleid ? null : handleClickStyle(bg, e))}>
			<div className={styles.imageContainer}>
				<Image className={currentStyleId === bg.styleid ? styles.selected : ""} style={{borderColor: selectedColor}} size="large" src={bg.icon}></Image>
				<div className={styles.imageLabel} style={{color: labelColor}}>{bg.name}</div>
			</div>
		</li>)}
	</ul>
</div>
);

ChooseStyle.defaultProps = {
	selectedColor: "#333",
	labelColor: "#333",
	handleClickStyle: () => {}
};

export default React.memo(ChooseStyle);

ChooseStyle.propTypes = {
	currentStyleId: PropTypes.number,
	stylesList: PropTypes.array,
	handleClickStyle: PropTypes.func,
	selectedColor: PropTypes.string,
	labelColor: PropTypes.string,
	handleClickBaseStyle: PropTypes.func,
};
