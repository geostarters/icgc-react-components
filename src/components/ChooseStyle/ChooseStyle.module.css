.styleContainer {
    max-height: 422px;
    overflow: auto;
}

ul {
    list-style: none;
    padding: 0px;
    margin: 0.15em 0;
}
img.selected { 
    border: 0.2em solid;
    padding: 0;
}

img {
    cursor: pointer;
    padding: 0.15em;
    height: 40px !important;
}

li {
    position: relative;
}

.imageContainer {
    position: relative;
    width: max-content;
}

.imageLabel {
    position: absolute;
    top: 50%;
    right: 10px;
    transform: translate(0%, -50%);
    font-weight: bold;
    text-shadow: 1.5px 1.5px 0 #ffffff, 1.5px -1.5px 0 #ffffff, -1.5px 1.5px 0 #ffffff, -1.5px -1.5px 0 #ffffff, 1.5px 0px 0 #ffffff, 0px 1.5px 0 #ffffff, -1.5px 0px 0 #ffffff, 0px -1.5px 0 #ffffff;
}
