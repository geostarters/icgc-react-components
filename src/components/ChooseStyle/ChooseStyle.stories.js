//@flow

import React from "react";
import { action } from "@storybook/addon-actions";

import ChooseStyle from "./ChooseStyle";

import "semantic-ui-css/semantic.min.css";

import styleList from "../../stories/styleList.json";

export default {
	title: "ChooseStyle",
	component: ChooseStyle
};

const Template = args => (<ChooseStyle {...args}/>);

export const _ChooseStyle = Template.bind({});
_ChooseStyle.storyName = "ChooseStyle";
_ChooseStyle.args = {
	stylesList: styleList
};

export const ChooseStyleSelected = Template.bind({});
ChooseStyleSelected.storyName = "ChooseStyle selected";
ChooseStyleSelected.args = {
	stylesList: styleList,
	currentStyleId: 0
};

export const ChooseStyleWithStyleAndAction = Template.bind({});
ChooseStyleWithStyleAndAction.storyName = "ChooseStyle with style and action";
ChooseStyleWithStyleAndAction.args = {
	stylesList: styleList,
	currentStyleId: 1,
	selectedColor: "#ff0000",
	labelColor: "#0000ff",
	handleClickStyle: action("onChange")
};
