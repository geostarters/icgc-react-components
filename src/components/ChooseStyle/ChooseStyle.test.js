// @flow

import React from "react";
import ChooseStyle from "./ChooseStyle";
import  { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import { Image } from "semantic-ui-react";

describe("ContainerChooseStyle Component", () => {

	const mockProps = {
		currentStyleId: 0,
		stylesList:[
			{
				"name": "Clar",
				"styleid": 0,
				"style": "https://geoserveis.icgc.cat/contextmaps/positron.json",
				"icon": "https://openicgc.github.io/img/positron.png"
			},
			{
				"name": "Orto",
				"styleid": 1,
				"style": "https://geoserveis.icgc.cat/contextmaps/hibrid.json",
				"icon": "https://openicgc.github.io/img/orto.png"
			}],
		handleClickStyle: jest.fn(),
		selectedColor: "#FF0000"

	};
	const component = shallow(<ChooseStyle {...mockProps}></ChooseStyle>);

	it("renders correct properties", () => {

		expect(toJSON(component)).toMatchSnapshot();

	});

	it("has only 2 list li elements", () => {

		expect(component.find("li")).toHaveLength(2);

	});

	it("has the correct color in the selected li element", () => {

		expect(component.find(Image).get(0).props.style).toHaveProperty("borderColor", mockProps.selectedColor);

	});

	it("calls the click handler after clicking a li element", () => {

		component.find("li").at(1).simulate("click");
		expect(mockProps.handleClickStyle).toBeCalled();

	});


});
