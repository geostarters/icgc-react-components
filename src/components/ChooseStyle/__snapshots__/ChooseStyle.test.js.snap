// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContainerChooseStyle Component renders correct properties 1`] = `
<div
  className="styleContainer"
>
  <ul>
    <li
      key="0"
      onClick={[Function]}
    >
      <div
        className="imageContainer"
      >
        <Image
          as="img"
          className="selected"
          size="large"
          src="https://openicgc.github.io/img/positron.png"
          style={
            Object {
              "borderColor": "#FF0000",
            }
          }
          ui={true}
        />
        <div
          className="imageLabel"
          style={
            Object {
              "color": "#333",
            }
          }
        >
          Clar
        </div>
      </div>
    </li>
    <li
      key="1"
      onClick={[Function]}
    >
      <div
        className="imageContainer"
      >
        <Image
          as="img"
          className=""
          size="large"
          src="https://openicgc.github.io/img/orto.png"
          style={
            Object {
              "borderColor": "#FF0000",
            }
          }
          ui={true}
        />
        <div
          className="imageLabel"
          style={
            Object {
              "color": "#333",
            }
          }
        >
          Orto
        </div>
      </div>
    </li>
  </ul>
</div>
`;
